meilisearch-http/src/routes/search.rs

```diff
@@ -53,45 +54,75 @@ pub async fn search_with_url_query(ctx: Request<Data>) -> SResult<Response> {
         search_builder.limit(limit);
     }
 
-    if let Some(attributes_to_retrieve) = query.attributes_to_retrieve {
-        for attr in attributes_to_retrieve.split(',') {
-            search_builder.add_retrievable_field(attr.to_string());
+    let available_attributes = schema.displayed_name();
+    let mut restricted_attributes: HashSet<&str>;
+    match &query.attributes_to_retrieve {
+        Some(attributes_to_retrieve) => {
+            let attributes_to_retrieve: HashSet<&str> = attributes_to_retrieve.split(',').collect();
```

Comment:
```suggestion
            let attributes_to_retrieve = attributes_to_retrieve.split(',').collect();
```

Reply:
I must give to attributes_to_retrieve a type at that point.

Reply:
Ho, yes, my mistake, I thought it was `restricted_attributes` sorry.